File: Famous Algorithms/towerofHanoi.py
def towerofHanoi(n, fr, to, aux):

	if n==1:
		print("Moving the disk 1 from " , fr , " to " , to , " rod.")

	else:
		towerofHanoi(n-1, fr, aux, to)
		print("Moving the disk ", n , " from " , fr , " to " , to , " rod.")
		towerofHanoi(n-1, aux, to, fr)

towerofHanoi(4, 'x', 'y', 'z')

File: Famous Algorithms/towerofHanoi.java
class Main{

  public void tOH(int n, char from, char to, char aux){

		if(n==1){
			System.out.println("Moving the disk 1 from " + from + " to " + to + " rod.");
		}

		else{
			tOH(n-1, from, aux, to);
			System.out.println("Moving the disk " + n + " from " + from + " to " + to + " rod.");
			tOH(n-1, aux, to , from);
		}

	}

	public static void main(String[] args){
    Main p = new Main();
    p.tOH(4, 'x', 'y', 'z');

	}

	
}


//time complexity is O(2^N)
//space complexity is O(N)